library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -726,15 +722,15 @@ applyEquation term rule = fmap (either id Success) $ runExceptT $ do
             "Equation with existentials: " <> Text.pack (show rule)
     -- immediately cancel if not preserving definedness
     unless (null rule.computedAttributes.notPreservesDefinednessReasons) $
-        throwE RuleNotPreservingDefinedness
+        throwE (Failure RuleNotPreservingDefinedness)
```

Comment:
I would suggest throwing the failure directly here
```suggestion
        throwE RuleNotPreservingDefinedness
```

and changing

```haskell
applyEquation term rule = fmap (either id Success) $ runExceptT $ do
```

to

 ```haskell
applyEquation term rule = fmap (either Failure Success) $ runExceptT $ do
```

Reply:
Good idea!